crates/chainio/clients/avsregistry/src/reader.rs

```diff
@@ -652,15 +655,442 @@ impl AvsRegistryChainReader {
+    /// * `operator_id` - The operator's identifier.
+    /// * `quorum_number` - The quorum number.
+    ///
+    /// # Returns
+    ///
+    /// * `Vec<StakeUpdate>` - A vector containing all stake updates.
+    pub async fn get_stake_history(
+        &self,
+        operator_id: B256,
+        quorum_number: u8,
+    ) -> Result<Vec<StakeUpdate>, AvsRegistryError> {
+        let provider = get_provider(&self.provider);
+
+        let contract_stake_registry = StakeRegistry::new(self.stake_registry_addr, provider);
+
+        let stake_update_vec = contract_stake_registry
+            .getStakeHistory(operator_id, quorum_number)
+            .call()
+            .await?
+            ._0;
+
+        Ok(stake_update_vec)
+    }
+
+    /// Returns the most recent stake update for a given operator and quorum.
+    ///
+    /// # Arguments
+    ///
+    /// * `operator_id` - The operator's identifier.
+    /// * `quorum_number` - The quorum number.
+    ///
+    /// # Returns
+    ///
+    /// * [`StakeUpdate`] - The latest stake update.
+    pub async fn get_latest_stake_update(
+        &self,
+        operator_id: B256,
+        quorum_number: u8,
+    ) -> Result<StakeUpdate, AvsRegistryError> {
+        let provider = get_provider(&self.provider);
+
+        let contract_stake_registry = StakeRegistry::new(self.stake_registry_addr, provider);
+
+        let stake_update = contract_stake_registry
+            .getLatestStakeUpdate(operator_id, quorum_number)
+            .call()
+            .await?
+            ._0;
+
+        Ok(stake_update)
+    }
+
+    /// Returns the stake update at a specific index for a given operator and quorum.
+    ///
+    /// # Arguments
+    ///
+    /// * `quorum_number` - The quorum number.
+    /// * `operator_id` - The operator's identifier.
+    /// * `index` - The index in the stake history array.
+    ///
+    /// # Returns
+    ///
+    /// * [`StakeUpdate`] - The stake update at the specified index.
+    pub async fn get_stake_update_at_index(
+        &self,
+        quorum_number: u8,
+        operator_id: B256,
+        index: U256,
+    ) -> Result<StakeUpdate, AvsRegistryError> {
+        let provider = get_provider(&self.provider);
+
+        let contract_stake_registry = StakeRegistry::new(self.stake_registry_addr, provider);
+
+        let stake_update = contract_stake_registry
+            .getStakeUpdateAtIndex(quorum_number, operator_id, index)
+            .call()
+            .await?
+            ._0;
+
+        Ok(stake_update)
+    }
+
+    /// Returns the stake of an operator for a given quorum at a specific block number.
+    ///
+    /// # Arguments
+    ///
+    /// * `operator_id` - The operator's identifier.
+    /// * `quorum_number` - The quorum number.
+    /// * `block_number` - The block number at which to retrieve the stake.
+    ///
+    /// # Returns
+    ///
+    /// * [`U96`] - The stake at the specified block number.
+    pub async fn get_stake_update_at_block_number(
```

Comment:
Since this calls `getStakeAtBlockNumber`, let's call it `get_stake_at_block_number`